packages/components/src/navigator/navigator-link/component.tsx

```diff
@@ -0,0 +1,72 @@
+/**
+ * External dependencies
+ */
+// eslint-disable-next-line no-restricted-imports
+import type { Ref } from 'react';
+
+/**
+ * WordPress dependencies
+ */
+import { useCallback } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import {
+	contextConnect,
+	useContextSystem,
+	WordPressComponentProps,
+} from '../../ui/context';
+import { View } from '../../view';
+import useNavigator from '../use-navigator';
+import type { NavigatorLinkProps } from '../types';
+
+const defaultAttributeName = 'id';
+const defaultSelectorFactory = ( attrName: string, attrValue: string ) =>
+	`[${ attrName }="${ attrValue }"]`;
+
+function NavigatorLink(
+	props: WordPressComponentProps< NavigatorLinkProps, 'a' >,
+	forwardedRef: Ref< any >
+) {
+	const {
+		path,
+		children,
+		onClick,
+		as = 'button',
+		attributeName = defaultAttributeName,
+		selectorFactory = defaultSelectorFactory,
```

Comment:
should these attributeName selectorFactory be props? Why should these be configurable?

Reply:
I thought it would be a good way to allow people to keep using `NavigatorLink` while allowing them to customise this aspect of the component.

E.g. — as a consumer of `Navigator`, I add a bunch of `NavigatorLink`. I then realise than `NavigatorLink` uses `id` by default, but that results in duplicated IDs in my page (since I have more than one `NavigatorLink` with the same `path`. By passing different `attributeName`s, I would be able to avoid the issue altogether.

The alternative would be to remove these 2 props, and assume that if a consumer of `NavigatorLink` has a similar issue, they can just look at how the component works and create their own custom version consuming `useNavigator` directly. WDYT?

Reply:
I'm always on the conservative side in general :P so I'd definitely lean towards avoiding the customization for now but wouldn't be against adding later if the use-case become more visible.

Reply:
For now, I decided to remove `selectorProp` in https://github.com/WordPress/gutenberg/pull/37223/commits/df584dea5e378a9e8c325528da6d8f0caa878181 and keep `attributeName` (for the reasons explained above)